### PR TITLE
Completed unit-tests-jest exercise - Eli Bales

### PR DESCRIPTION
**Pull request by e-bales**

### CODE REVIEW

File: unit-tests-jest/numbers.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * `toDollars(11.341)` returns `$11.34`.
+ */
+export function toDollars(amount) {
+  if (Number.isNaN(Number(amount))) {
+    return 'Please enter an integer or float.';
+  }
+  if (Number.isInteger(amount)) {
+    return `$${amount}.00`;
+  } else {
+    return `$${amount.toFixed(2)}`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:24]: `toFixed` works for integers, too.

File: unit-tests-jest/numbers.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Returns a new array that contains only the even numbers in the given array.
+ */
+export function evenNumbers(numbers) {
+  if (!Array.isArray(numbers)) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:5]: Nice edge case check, but you should `throw new Error(message);` instead of returning a string.

You can test that using the `toThrow` matcher.

File: unit-tests-jest/numbers.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ * It works for both integer and fractional numbers.
+ * That is, `toDollars(12)` returns `$12.00` and
+ * `toDollars(11.341)` returns `$11.34`.
+ */
+export function toDollars(amount) {
+  if (Number.isNaN(Number(amount))) {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:18]: Nice edge case check. See above.

You have many very nice edge case checks, but they should all throw.

File: unit-tests-jest/numbers.test.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  });
+});
+
+describe('evenNumbers', () => {
+  it('returns an error message if not provided an array', () => {
+    const numbers = 'A String';
+    const result = evenNumbers(numbers);
+    expect(result).toEqual('Please provide an array.');
+  });
+});
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:34]: I like these edge case tests. With `describe`, we usually put all of them together that test the same function.

It's like "describe evenNumbers -- it returns the even numbers of an array; it returns an error message if not an array" and so on.

The code looks like this:
```
describe('evenNumbers', () => {
  it('returns the even numbers of an array', () => {...});
  it('returns an error message if not provided an array', () => {...});
  ...
});
```

File: unit-tests-jest/numbers.test.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    const result = toDollars(amount);
+    expect(result).toEqual('$0.56');
+  });
+});
+
+describe('toDollars', () => {
+  it('returns an error message given a string instead of an int/float', () => {
+    const amount = 'A String';
+    const result = toDollars(amount);
+    expect(result).toEqual('Please enter an integer or float.');
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:65]: With exceptions, you would test it like this:
```
expect(() => toDollars(amount)).toThrow('Please enter an integer or float.');
```

File: unit-tests-jest/numbers.test.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    expect(result).toEqual({ a: 2, b: '2', c: 10 });
+  });
+});
+
+describe('multiplyBy', () => {
+  it('returns a MODIFIED object, not a new object', () => {
+    const object = { a: 1, b: '2', c: 5 };
+    const multiplier = 2;
+    const result = multiplyBy(object, multiplier);
+    expect(result).toEqual(object);
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:151]: You need to use `toBe` here, so that it tests reference equality.

**Review by RobertGardner — APPROVED** [review]
Very thorough testing. Nice!
Some suggestions on how to improve the tests.
